python/src/python-module.cc

```diff
@@ -0,0 +1,75 @@
+#include <Python.h>
+
+#include "internal/eval.hh"
+#include "internal/ptr.hh"
+
+#include <eval.hh>
+#include <globals.hh>
+#include <shared.hh>
+
+namespace nix::python {
+
+#define _public_ __attribute__((visibility("default")))
+
+PyObject * ThrownNixError = nullptr;
+PyObject * NixError = nullptr;
+
+static PyMethodDef NixMethods[] = {
+    {"eval", (PyCFunction) eval, METH_VARARGS | METH_KEYWORDS, "Eval nix expression"}, {NULL, NULL, 0, NULL}};
+
+static struct PyModuleDef nixmodule = {
+    PyModuleDef_HEAD_INIT, "nix", "Nix expression bindings",
+    -1, /* size of per-interpreter state of the module,
+           or -1 if the module keeps state in global variables. */
+    NixMethods};
+
+extern "C" _public_ PyObject * PyInit_nix(void)
+{
+    // By default, Nix sets the build-hook to be "$(readlink /proc/self/exe) __build-remote", expecting the current
```

Comment:
This is a very ugly work-around. Maybe we can fix this in a better way? Require an argument to `initNix` maybe?

Reply:
> `initNix`

It should only have to initialize the things it uses, and `initNix` should only be called by `libmain` consumers. Needs this WIP though:
 - #7732 
 
I don't think that solves the need for `buildHook` initialization yet, but I do think something needs to be done for `buildHook`, and it should be done `libstore` as part of regular initialization or some helper function.